src/main/plugins/org.dita.pdf2/xsl/fo/loc.xsl

```diff
@@ -0,0 +1,178 @@
+          </fo:block>
+        </fo:flow>
+      </fo:page-sequence>
+    </xsl:if>
+  </xsl:template>
+
+  <xsl:template name="createLOCHeader">
+    <fo:block xsl:use-attribute-sets="__lotf__heading" id="{$id.lot}">
+      <fo:marker marker-class-name="current-header">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'List of Changes'"/>
+        </xsl:call-template>
+      </fo:marker>
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'List of Changes'"/>
+      </xsl:call-template>
+    </fo:block>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class, ' bookmap/bookmeta ')]/*[contains(@class, ' topic/critdates ')]">
+
+    <!-- Create a list of all book releases -->
+    <xsl:variable name="bookReleaseDates" select="*[contains (@class, ' topic/created ')]/@date |
+                                                  *[contains (@class, ' topic/revised ')]/@modified"/>
+    <xsl:variable name="bookReleaseDateString">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'Book Release Date'"/>
+      </xsl:call-template>
+    </xsl:variable>
+    <xsl:variable name="colonSymbol">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'ColonSymbol'"/>
+      </xsl:call-template>
+    </xsl:variable>
+    <xsl:variable name="separator">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'figure-number-separator'"/>
+      </xsl:call-template>
+    </xsl:variable>
+
+    <xsl:for-each select="$bookReleaseDates">
+      <xsl:variable name="previousDateIndex" select="position() - 1"/>
+      <xsl:variable name="previousDate" select="xs:date($bookReleaseDates[$previousDateIndex])"/>
+      <xsl:variable name="date" select="xs:date(.)"/>
+      <fo:block>
+        <fo:table xsl:use-attribute-sets="releaseManagementTable">
+          <xsl:call-template name="selectAtts"/>
+          <xsl:call-template name="globalAtts"/>
+          <fo:table-body>
+            <fo:table-row xsl:use-attribute-sets="tbody.row">
+              <fo:table-cell xsl:use-attribute-sets="tbody.row.entry">
+                <fo:block xsl:use-attribute-sets="tbody.row.entry__content">
+                  <xsl:value-of select="$bookReleaseDateString"/>
+                  <xsl:value-of select="$colonSymbol"/>
+                  <xsl:value-of select="$separator"/>
+                  <xsl:value-of select="format-date(., $dateFormat)"/>
+                </fo:block>
+              </fo:table-cell>
+            </fo:table-row>
+            <xsl:for-each select="$change-items">
+              <xsl:if test="./*[contains (@class, ' relmgmt-d/change-completed ')] != ''">
+                <xsl:variable name="change-completed" select="xs:date(./*[contains (@class, ' relmgmt-d/change-completed ')])"/>
+                <xsl:if test="($previousDate &lt;= $change-completed)

```

Comment:
The logic is hard to follow. Filtering should be done earlier if it's needed. The idea that previous date is used here is something I just don't understand. Could this not be handled with sorting at a higher level?